File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/StrikeSchedule.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class StrikeSchedule : Schedule
    {
        public StrikeSchedule(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode buyerNode = xmlNode.SelectSingleNode("buyer");
            
            if (buyerNode != null)
            {
                if (buyerNode.Attributes["href"] != null || buyerNode.Attributes["id"] != null) 
                {
                    if (buyerNode.Attributes["id"] != null) 
                    {
                        buyerIDRef_ = buyerNode.Attributes["id"].Value;
                        IdentifiedPayerReceiver ob = new IdentifiedPayerReceiver(buyerNode);
                        IDManager.SetID(buyerIDRef_, ob);
                    }
                    else if (buyerNode.Attributes["href"] != null)
                    {
                        buyerIDRef_ = buyerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        buyer_ = new IdentifiedPayerReceiver(buyerNode);
                    }
                }
                else
                {
                    buyer_ = new IdentifiedPayerReceiver(buyerNode);
                }
            }
            
        
            XmlNode sellerNode = xmlNode.SelectSingleNode("seller");
            
            if (sellerNode != null)
            {
                if (sellerNode.Attributes["href"] != null || sellerNode.Attributes["id"] != null) 
                {
                    if (sellerNode.Attributes["id"] != null) 
                    {
                        sellerIDRef_ = sellerNode.Attributes["id"].Value;
                        IdentifiedPayerReceiver ob = new IdentifiedPayerReceiver(sellerNode);
                        IDManager.SetID(sellerIDRef_, ob);
                    }
                    else if (sellerNode.Attributes["href"] != null)
                    {
                        sellerIDRef_ = sellerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        seller_ = new IdentifiedPayerReceiver(sellerNode);
                    }
                }
                else
                {
                    seller_ = new IdentifiedPayerReceiver(sellerNode);
                }
            }
            
        
        }
        
    
        #region Buyer_
        private IdentifiedPayerReceiver buyer_;
        public IdentifiedPayerReceiver Buyer_
        {
            get
            {
                if (this.buyer_ != null)
                {
                    return this.buyer_; 
                }
                else if (this.buyerIDRef_ != null)
                {
                    buyer_ = IDManager.getID(buyerIDRef_) as IdentifiedPayerReceiver;
                    return this.buyer_; 
                }
                else
                {
                      return this.buyer_; 
                }
            }
            set
            {
                if (this.buyer_ != value)
                {
                    this.buyer_ = value;
                }
            }
        }
        #endregion
        
        public string buyerIDRef_ { get; set; }
        #region Seller_
        private IdentifiedPayerReceiver seller_;
        public IdentifiedPayerReceiver Seller_
        {
            get
            {
                if (this.seller_ != null)
                {
                    return this.seller_; 
                }
                else if (this.sellerIDRef_ != null)
                {
                    seller_ = IDManager.getID(sellerIDRef_) as IdentifiedPayerReceiver;
                    return this.seller_; 
                }
                else
                {
                      return this.seller_; 
                }
            }
            set
            {
                if (this.seller_ != value)
                {
                    this.seller_ = value;
                }
            }
        }
        #endregion
        
        public string sellerIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/StrikeSchedule.cpp
// StrikeSchedule.cpp 
#include "StrikeSchedule.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

StrikeSchedule::StrikeSchedule(TiXmlNode* xmlNode)
: Schedule(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //buyerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* buyerNode = xmlNode->FirstChildElement("buyer");

   if(buyerNode){buyerIsNull_ = false;}
   else{buyerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- buyerNode , address : " << buyerNode << std::endl;
   #endif
   if(buyerNode)
   {
      if(buyerNode->Attribute("href") || buyerNode->Attribute("id"))
      {
          if(buyerNode->Attribute("id"))
          {
             buyerIDRef_ = buyerNode->Attribute("id");
             buyer_ = boost::shared_ptr<IdentifiedPayerReceiver>(new IdentifiedPayerReceiver(buyerNode));
             buyer_->setID(buyerIDRef_);
             IDManager::instance().setID(buyerIDRef_,buyer_);
          }
          else if(buyerNode->Attribute("href")) { buyerIDRef_ = buyerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { buyer_ = boost::shared_ptr<IdentifiedPayerReceiver>(new IdentifiedPayerReceiver(buyerNode));}
   }

   //sellerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* sellerNode = xmlNode->FirstChildElement("seller");

   if(sellerNode){sellerIsNull_ = false;}
   else{sellerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- sellerNode , address : " << sellerNode << std::endl;
   #endif
   if(sellerNode)
   {
      if(sellerNode->Attribute("href") || sellerNode->Attribute("id"))
      {
          if(sellerNode->Attribute("id"))
          {
             sellerIDRef_ = sellerNode->Attribute("id");
             seller_ = boost::shared_ptr<IdentifiedPayerReceiver>(new IdentifiedPayerReceiver(sellerNode));
             seller_->setID(sellerIDRef_);
             IDManager::instance().setID(sellerIDRef_,seller_);
          }
          else if(sellerNode->Attribute("href")) { sellerIDRef_ = sellerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { seller_ = boost::shared_ptr<IdentifiedPayerReceiver>(new IdentifiedPayerReceiver(sellerNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<IdentifiedPayerReceiver> StrikeSchedule::getBuyer()
{
   if(!this->buyerIsNull_){
        if(buyerIDRef_ != ""){
             return boost::shared_static_cast<IdentifiedPayerReceiver>(IDManager::instance().getID(buyerIDRef_));
        }else{
             return this->buyer_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<IdentifiedPayerReceiver>();
   }
}
boost::shared_ptr<IdentifiedPayerReceiver> StrikeSchedule::getSeller()
{
   if(!this->sellerIsNull_){
        if(sellerIDRef_ != ""){
             return boost::shared_static_cast<IdentifiedPayerReceiver>(IDManager::instance().getID(sellerIDRef_));
        }else{
             return this->seller_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<IdentifiedPayerReceiver>();
   }
}
}

